plotly/statistical_charts.py

```diff
@@ -0,0 +1,439 @@
+        - **Violin Plots:** Explore distribution shapes and density estimates.  
+        - **Histograms:** Analyze frequency distributions and experiment with different bin sizes.  
+        - **Error Bars:** Visualize measurement uncertainties and variability.
+
+        We will use the built-in [tips dataset](https://github.com/plotly/datasets/blob/master/tips.csv) from Plotly for our examples. This dataset contains real-world information about restaurant bills, tips, and customer attributes, making it perfect for exploring statistical visualizations. Here are the first few rows of this dataset displayed below:
+        """
+    )
+    return
+
+
+@app.cell
+def _(px):
+    tips = px.data.tips()
+    tips.head()
+    return (tips,)
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""
+        ## Box Plots
+
+        Box plots help you visualize the five-number summary (minimum, first quartile, median, third quartile, and maximum) along with potential outliers. In this example, you’ll explore how the total bill varies by different grouping variables like day, time, or sex.
+        """
+    )
+    return
+
+
+@app.cell
+def _(mo):
+    dropdown = mo.ui.dropdown(
+        options=['day', 'time', 'sex'],
+        value='day',
+    )
+    return (dropdown,)
+
+
+@app.cell
+def _(dropdown, mo):
+    mo.hstack([dropdown, mo.md(f"Selected Variable: {dropdown.value}")])
+    return
+
+
+@app.cell
+def _(dropdown, update_boxplot):
+    update_boxplot(dropdown.value)
+    return
+
+
+@app.cell
+def _(mo):
+    mo.md(
+        r"""
+        Explore how customer spending changes using this interactive box plot of total bills. By default, you're seeing how bills vary by day of the week. This box plot gives a clear look at how total bill amounts vary by day of the week. It’s immediately noticeable that Saturday and Sunday have higher medians and more extreme outliers compared to weekdays like Thursday and Friday. This makes intuitive sense as weekends are peak times for dining out, leading to larger group sizes, longer stays, and potentially more expensive meals, which contribute to higher bills.
+
+        A couple of interesting insights:
+
+        * Sunday shows a noticeably wider spread, hinting that people might either go for cheap brunches or big family dinners.
+        * Friday seems relatively modest in terms of both spread and outliers. Perhaps people are grabbing quicker meals or just transitioning into the weekend.
+        * Thursday is visibly lower, indicating a quieter dining day with fewer large bills.
+
+        If you switch the dropdown to group by time of day or sex, you might notice:
+
+        * Dinner bills tend to be higher than lunch.
+        * Male diners often have slightly higher bills.
+
+        This kind of visualization really helps tell the story of human behavior through data and their eating habits throughout the week.
+        """
+    )
+    return
+
+
+@app.cell
```

Comment:
```suggestion
@app.cell(hide_code=True)
```

Can we hide all markdown cells listed in the notebook?